### PR TITLE
feat(translator): implement buffer limit support in CTP

### PR DESCRIPTION
**Pull request by yaelSchechter**
**What type of PR is this?**
Implement buffer limit in ClientTrafficPolicy
Follow up of https://github.com/envoyproxy/gateway/pull/2805

Fixes https://github.com/envoyproxy/gateway/issues/2600

### CODE REVIEW

File: internal/gatewayapi/clienttrafficpolicy.go [hunk 1]
```diff
@@ -693,7 +694,18 @@ func translateListenerConnection(connection *egv1a1.Connection, httpIR *ir.HTTPL
...
+		irConnection.ConnectionLimit = irConnectionLimit
+	}
+
+	if connection.BufferLimit != nil {
+		bufferLimit, ok := connection.BufferLimit.AsInt64()
+		if !ok {
+			return fmt.Errorf("invalid BufferLimit value %s", connection.BufferLimit.String())
+		}
+		if bufferLimit < 0 || bufferLimit > math.MaxUint32 {
+			return fmt.Errorf("BufferLimit value %d out of range", bufferLimit)
```
[the remark this answers is not in the document]
Comment (yaelSchechter) [reply internal/gatewayapi/clienttrafficpolicy.go]: done, thanks

**Review by guydc — APPROVED** [review]

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by arkodg — APPROVED** [review]
LGTM thanks 🚀  !

**Pull request merged**